src/AppKit/NSOpenGLPixelFormat.cs

```diff
@@ -54,67 +54,94 @@ public NSOpenGLPixelFormat (NSOpenGLPixelFormatAttribute[] attribs) : base (NSOb
-				case NSOpenGLPixelFormatAttribute.RemotePixelBuffer:
-				case NSOpenGLPixelFormatAttribute.AuxDepthStencil:
-				case NSOpenGLPixelFormatAttribute.ColorFloat:
-				case NSOpenGLPixelFormatAttribute.Multisample:
-				case NSOpenGLPixelFormatAttribute.Supersample:
-				case NSOpenGLPixelFormatAttribute.SampleAlpha:
-				case NSOpenGLPixelFormatAttribute.AllowOfflineRenderers:
-				case NSOpenGLPixelFormatAttribute.AcceleratedCompute:
-				case NSOpenGLPixelFormatAttribute.MPSafe:
-					list.Add (v);
-					break;
-					
-				case NSOpenGLPixelFormatAttribute.AuxBuffers:
-				case NSOpenGLPixelFormatAttribute.ColorSize:
-				case NSOpenGLPixelFormatAttribute.AlphaSize:
-				case NSOpenGLPixelFormatAttribute.DepthSize:
-				case NSOpenGLPixelFormatAttribute.StencilSize:
-				case NSOpenGLPixelFormatAttribute.AccumSize:
-				case NSOpenGLPixelFormatAttribute.RendererID:
-				case NSOpenGLPixelFormatAttribute.ScreenMask:
+					case NSOpenGLPixelFormatAttribute.RemotePixelBuffer:
+					case NSOpenGLPixelFormatAttribute.AuxDepthStencil:
+					case NSOpenGLPixelFormatAttribute.ColorFloat:
+					case NSOpenGLPixelFormatAttribute.Multisample:
+					case NSOpenGLPixelFormatAttribute.Supersample:
+					case NSOpenGLPixelFormatAttribute.SampleAlpha:
+					case NSOpenGLPixelFormatAttribute.AllowOfflineRenderers:
+					case NSOpenGLPixelFormatAttribute.AcceleratedCompute:
+					case NSOpenGLPixelFormatAttribute.MPSafe:
+						list.Add ((uint) v);
+						break;
+
+					case NSOpenGLPixelFormatAttribute.AuxBuffers:
+					case NSOpenGLPixelFormatAttribute.ColorSize:
+					case NSOpenGLPixelFormatAttribute.AlphaSize:
+					case NSOpenGLPixelFormatAttribute.DepthSize:
+					case NSOpenGLPixelFormatAttribute.StencilSize:
+					case NSOpenGLPixelFormatAttribute.AccumSize:
+					case NSOpenGLPixelFormatAttribute.RendererID:
+					case NSOpenGLPixelFormatAttribute.ScreenMask:
 
 					// not listed in the docs, but header file implies it
-				case NSOpenGLPixelFormatAttribute.SampleBuffers:
-				case NSOpenGLPixelFormatAttribute.Samples:
-				case NSOpenGLPixelFormatAttribute.VirtualScreenCount:
-					list.Add (v);
-					i++;
-					if (i >= args.Length)
-						throw new ArgumentException ("Attribute " + v + " needs a value");
-					if (args[i] is NSOpenGLPixelFormatAttribute)
-						list.Add ((NSOpenGLPixelFormatAttribute) args [i]);
-					else
-						list.Add ((NSOpenGLPixelFormatAttribute) (int)args [i]);
-					break;
-				}
+					case NSOpenGLPixelFormatAttribute.SampleBuffers:
+					case NSOpenGLPixelFormatAttribute.Samples:
+					case NSOpenGLPixelFormatAttribute.VirtualScreenCount:
+					case NSOpenGLPixelFormatAttribute.OpenGLProfile:
+						list.Add ((uint)v);
+						i++;
+						if (i >= args.Length)
+							throw new ArgumentException ("Attribute " + v + " needs a value");
+						list.Add ((uint) (int)args [i]);
+						if (i == args.Length - 1)
+							throw new ArgumentException ("The attributes array must end with a 0");
```

Comment:
Should I just make this a CWL instead (Rolf said something about breaking changes)